src/subject_name/mod.rs

```diff
@@ -203,19 +203,17 @@ enum Subtrees {
 
 pub(crate) struct NameIterator<'a> {
     subject_alt_name: Option<untrusted::Reader<'a>>,
-    subject_directory_name: Option<untrusted::Input<'a>>,
+    directory_name: bool,
```

Comment:
I think this would be clearer named `include_directory_name` or `consider_directory_name` (sorry to bikeshed...). 

A two-variant enum would be even clearer IMO but if you think that's overkill for this internal use-case I wouldn't argue.